core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -195,15 +191,15 @@ public ValidationResponse validateChallenge(ValidationRequest validationRequest)
   }
 
   public String validateChallenge(String challengeXdr)
-      throws IOException, InvalidSep10ChallengeException, URISyntaxException {
+      throws IOException, InvalidSep10ChallengeException {
     debug("Parse challenge string.");
     Sep10Challenge.ChallengeTransaction challenge =
         Sep10Challenge.readChallengeTransaction(
             challengeXdr,
             serverAccountId,
             new Network(appConfig.getStellarNetworkPassphrase()),
-            sep10Config.getHomeDomain(),
-            getDomainFromURI(appConfig.getHostUrl()));
+            sep10Config.getWebAuthDomain(),
+            getDomainFromURL(appConfig.getHostUrl()));
```

Comment:
This should be switched. The `home_domain` query parameter in SEP-10 GET should match `host_url`, not `web_auth_domain`.

```suggestion
            getDomainFromURL(appConfig.getHostUrl()),
            sep10Config.getWebAuthDomain());
```

Reply:
👍 Good catch. The order is switched now.